### PR TITLE
Send 204 when Campaign doesn't have keywords

### LINKED ISSUE #189: Send message TypeError: Cannot read property 'externalSignupMenu' of undefined
Requests from Blink are triggering this on staging and my hunch is that we're receiving requests for Campaigns that aren't running on Gambit.

In a Gambit Campaigns `POST /signups` request (which will be replaced by posting to the Conversations `POST /send-message), we return a 422 if the target Campaign is [closed](https://github.com/DoSomething/gambit/blob/4.9.0/app/routes/signups.js#L82) or doesn't have any [published keywords](https://github.com/DoSomething/gambit/blob/4.9.0/app/routes/signups.js#L105).

### PR DESCRIPTION
**Pull request by rapala61**
# What does this PR do?
WIP

# Relevant issues.
Fixes #189

### CODE REVIEW

**Comment by aaronschachter** [comment]
This branch has been updated and ready for review, picking up where @rapala61 left off on bug fixing.  

* Fixes closed campaign check (I missed this in #184)
* Sends a 204 if a Campaign is found in the Gambit Campaigns API, but it doesn't have any keywords available

* Refactors `gambitCampaigns.isActiveCampaign` to `gambitCampaigns.isClosedCampaign`: It's clearer what a closed campaign is. An Phoenix Campaign with status `active` could have no keywords (which is what we've been talking about as an invalid Gambit Campaign, but should just start referring this scenario as whether a Campaign has keywords, or is available/running on Gambit )

The `TypeError` that gets thrown will be resolved once a fix for https://github.com/DoSomething/gambit/issues/990 is in place. @rapala61 I'm unable to assign this to you for review since you opened it, mind reviewing whenever you get a chance? I'll be jumping on https://github.com/DoSomething/gambit/issues/990 next.

**Comment by rapala61** [comment]
It looks good to me! But I can't approve myself so I'm going to make you judge yourself haha.

File: lib/middleware/send-message/campaign.js [hunk 1]
```diff
@@ -20,10 +20,13 @@ module.exports = function sendCampaignTemplate() {
         if (!campaign) {
           return helpers.sendResponseWithStatusCode(res, 404, 'Campaign not found.');
         }
-        if (campaign.isClosed) {
+        if (gambitCampaigns.isClosedCampaign(campaign)) {
           const err = new UnprocessibleEntityError('Campaign is closed.');
           return helpers.sendErrorResponse(res, err);
         }
+        if (campaign.keywords.length === 0) {
```
Comment (rapala61) [inline lib/middleware/send-message/campaign.js]: Just to over-document: This check is not triggering now until https://github.com/DoSomething/gambit/issues/990 gets fixed.

File: lib/middleware/send-message/campaign.js [hunk 2]
```diff
@@ -34,6 +37,7 @@ module.exports = function sendCampaignTemplate() {
         return req.conversation.setCampaign(campaign)
           .then(() => next())
           .catch(err => helpers.sendErrorResponse(res, err));
-      });
+      })
+      .catch(err => helpers.sendErrorResponse(res, err));
```
Comment (rapala61) [inline lib/middleware/send-message/campaign.js]: Over-documenting: The original error on issue #189 will get fixed with this catch.

**Pull request merged**